File: admin-web/src/api/dataDict.ts
import http from './http'

export interface DictItem {
  id: number
  category: string
  code: string
  label: string
  sort_order: number
  is_active: boolean
}

export const listDict = (category: string) =>
  http.get<DictItem[]>('/data-dict', { params: { category } }).then((r) => r.data)

File: backend/app/models/data_dict.py
from datetime import datetime

from sqlalchemy import DateTime, String, UniqueConstraint, func
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class DataDict(Base):
    """Generic data dictionary: category (e.g. skill/expense_type) → list of items."""

    __tablename__ = "data_dict"
    __table_args__ = (UniqueConstraint("category", "code", name="uq_dict_category_code"),)

    id: Mapped[int] = mapped_column(primary_key=True)
    category: Mapped[str] = mapped_column(String(64), index=True)
    code: Mapped[str] = mapped_column(String(64))
    label: Mapped[str] = mapped_column(String(128))
    sort_order: Mapped[int] = mapped_column(default=0)
    is_active: Mapped[bool] = mapped_column(default=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
